controllers/aiven_application/reconciler.go

```diff
@@ -113,6 +114,15 @@ func (r *AivenApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return fail(err)
 	}
 
+	if ok := annotations.HasDelete(application.GetAnnotations()); ok {
+		err = r.handleDeleteAnnotation(ctx, application, logger, ok)
+		if err != nil {
+			return fail(err)
```

Comment:
Her kan du nok med fordel sørge for å kalle `utils.LocalFail` med en beskrivende "reason" før du returnerer.